lychee-lib/src/client.rs

```diff
@@ -236,6 +239,11 @@ impl Client {
 
     pub async fn check_mail(&self, uri: &Uri) -> Status {
         let input = CheckEmailInput::new(vec![uri.as_str().to_owned()]);
+
+        // REVIEW: This panics with an out-of-bounds error
+        // if there are no results from check_email
+        // (previous versions returned false)
+        // these parenthesis also seem unnecesary
         let result = &(check_email(&input).await)[0];
```

Comment:
Unnecessary. This will not happen.
Read the source code here https://docs.rs/check-if-email-exists/0.8.21/src/check_if_email_exists/lib.rs.html#194-214

Reply:
The question is unnecessary?
I don't really see how that part of the code guarantees that the result will never be empty as it just maps over the input and it doesn't really answer why the parenthesis is necessary.
Since you aparetnly have found a sound way of reasoning that this can not panic I would appreciate if you could suggest a comment to go here that explains it.

Reply:
`input` is of type `CheckEmailInput`, which contains a vector of URL https://docs.rs/check-if-email-exists/0.8.21/src/check_if_email_exists/util/input_output.rs.html#77

`check_email` returns an `impl Future<output = Vec<CheckEmailOutput>>`, where the vector of output has same dimension as the vector of URL.

It's very obvious. As long as the input is non empty, the output is not. So it won't panic.
---
You can remove the parenthesis though.

Reply:
Ah, I didn't see the line above which creates the input because it had been cut off by the github diff and assumed that was supplied by the user, sorry for the confusion.